app/src/main/java/com/google/android/sambadocumentsprovider/mount/MountServerActivity.java

```diff
@@ -177,14 +215,57 @@ public void onSaveInstanceState(Bundle outState) {
 
     outState.putString(SHARE_PATH_KEY, mSharePathEditText.getText().toString());
     final boolean needsPassword = mNeedPasswordCheckbox.isChecked();
+
     outState.putBoolean(NEEDS_PASSWORD_KEY, needsPassword);
-    if (needsPassword) {
+    outState.putBoolean(AUTH_LAUNCH_KEY, isLaunchedForAuth);
+
+    if (needsPassword || isLaunchedForAuth) {
       outState.putString(DOMAIN_KEY, mDomainEditText.getText().toString());
       outState.putString(USERNAME_KEY, mUsernameEditText.getText().toString());
       outState.putString(PASSWORD_KEY, mPasswordEditText.getText().toString());
     }
   }
 
+  private void prepareForAuth(Intent authIntent) {
+    final Uri shareUri = Uri.parse(authIntent.getStringExtra(SHARE_URI_KEY));
+    final DocumentMetadata shareMetadata = DocumentMetadata.createShare(shareUri);
+
+    displayAuthUI(shareUri.toString());
+
+    mAuthChecker = new ShareManager.ShareMountChecker() {
```

Comment:
Don't save this to field. Create it when it's needed.

You're gonna launch different activities for different authentications, so it doesn't save anything.